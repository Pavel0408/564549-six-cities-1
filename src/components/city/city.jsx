import PropTypes from "prop-types";
import React, {PureComponent} from "react";

export class City extends PureComponent {
  constructor(props) {
    super(props);

    this.cityClickHandler = this.cityClickHandler.bind(this);
  }

  cityClickHandler(evt) {
    evt.preventDefault();
    this.props.cityClickHandler(this.props.city);
  }

  render() {
    const isActive = this.props.isActive;

    return <li className="locations__item">
      <a className={isActive ? `locations__item-link tabs__item tabs__item tabs__item--active` : `locations__item-link tabs__item tabs__item`} onClick={this.cityClickHandler}>
        <span>{this.props.city}</span>
      </a>
    </li>;
  }
}

City.propTypes = {
  cityClickHandler: PropTypes.func,
  city: PropTypes.string,
  isActive: PropTypes.bool
};
