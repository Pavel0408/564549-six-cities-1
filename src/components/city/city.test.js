import React from 'react';
import renderer from 'react-test-renderer';
import {City} from './city';

const activeCity = `Amsterdam`;
const cityClickHandle = () => {
};

describe(`testing City render`, () => {
  it(`City is render correctly`, () => {
    const tree = renderer.create(<City
      key={`city-${1}`}
      city = {activeCity}
      isActive = {true}
      cityClickHandle={cityClickHandle}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
