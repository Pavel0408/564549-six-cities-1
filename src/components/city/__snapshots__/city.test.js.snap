// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing City render City is render correctly 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item tabs__item--active"
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;
